Zend/zend_language_scanner.l

```diff
@@ -1109,12 +1109,29 @@ skip_escape_conversion:
 #define HEREDOC_USING_SPACES 1
 #define HEREDOC_USING_TABS 2
 
-static zend_bool strip_multiline_string_indentation(zval *zendlval, int newline, int indentation, zend_bool using_spaces)
+static zend_bool strip_multiline_string_indentation(zval *zendlval, int newline, int indentation, zend_bool using_spaces, zend_bool body_beginning)
 {
 	int len = Z_STRLEN_P(zendlval), new_len = len, i = 0, j = 0, skip, newline_count = 0;
 	char *copy = Z_STRVAL_P(zendlval);
 	zend_bool trailing_newline = 0;
 
+	// if we are NOT at the beginning of the heredoc, then skip up to (and including) the next new line
+	if (!body_beginning) {
+		while (j < len && copy[j] != '\r' && copy[j] != '\n') {
+			++j;
+		}
+
+		if (copy[j] == '\r') {
+			++j;
+		}
+
+		if (copy[j] == '\n') {
+			++j;
+		}
+
```

Comment:
Have you considered to merge the two `if` statements above?

Reply:
It would not work in case of `\r\n`.

Reply:
Oops, it's as @imbrish said. Sorry for the noise.